dcos/cosmospackage.py

```diff
@@ -704,3 +734,38 @@ def _v2_package_to_v1_package_json(package_info):
         del package_json["resource"]
 
     return package_json
+
+
+def _file_length(filename):
+    """
+    Gets the length of a file
+
+    :param filename: a path to a file
+    :type filename: str
+    :return: length of the file whose path is filename
+    :rtype: int
+    """
+    with util.open_file(filename, 'r+b') as file:
+        file.seek(0, 2)
+        return file.tell()
```

Comment:
You can use `https://docs.python.org/3.6/library/os.path.html#os.path.getsize`. When we move to now using chunked encoding please investigate if requests is able to populate the `content-length` automatically when you pass a file object.